server/handlers.go

```
@@ -105,27 +107,47 @@ func (svr *Server) handleGetOPGenericCommitment(w http.ResponseWriter, r *http.R
 	}
 	versionedCert := certs.NewVersionedCert(commitment, certVersion)
 
-	return svr.handleGetShared(r.Context(), w, versionedCert, commitments.OptimismGenericCommitmentMode)
+	return svr.handleGetShared(r.Context(), w, r, versionedCert, commitments.OptimismGenericCommitmentMode)
 }
 
 func (svr *Server) handleGetShared(
 	ctx context.Context,
 	w http.ResponseWriter,
+	r *http.Request,
 	versionedCert certs.VersionedCert,
 	mode commitments.CommitmentMode,
 ) error {
 	serializedCertHex := hex.EncodeToString(versionedCert.SerializedCert)
 	svr.log.Info("Processing GET request", "commitmentMode", mode,
 		"certVersion", versionedCert.Version, "serializedCert", serializedCertHex)
-	input, err := svr.sm.Get(ctx, versionedCert, mode)
+
+	l1InclusionBlockNum, err := parseCommitmentInclusionL1BlockNumQueryParam(r)
+	if err != nil {
+		err = NewGETError(
+			fmt.Errorf("invalid l1_block_number: %w", err),
+			versionedCert.Version, mode)
+		// the inclusion block query param is optional, but if it is provided and invalid, we return a 400 error
+		// to let the client know that they probably have a bug.
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return err
+	}
+	input, err := svr.sm.Get(
+		ctx,
+		versionedCert,
+		mode,
+		common.CertVerificationOpts{L1InclusionBlockNum: l1InclusionBlockNum},
+	)
 	if err != nil {
 		err = NewGETError(
 			fmt.Errorf("get request failed with serializedCert %v: %w", serializedCertHex, err),
-			versionedCert.Version,
-			mode,
-		)
-		if errors.Is(err, ErrNotFound) {
-			http.Error(w, err.Error(), http.StatusNotFound)
+			versionedCert.Version, mode)
+		var rbnRecencyCheckFailedErr eigendav2store.RBNRecencyCheckFailedError
+		if errors.As(err, &rbnRecencyCheckFailedErr) {
+			// We return a 418 TEAPOT error for any cert validation error.
```

though unrelated. Consider a case, that cert is valid, and recency is good. It is just the blob encoding has problem. In that case, hokulea has to deal with that in the derivation pipeline. Question, can proxy return such invalid encoding cert? 
Currently, proxy only returns the decoded data to hokulea.

---

Prob better to talk this one out using voice call.
1. wdym by "blob encoding has problem"? If a cert is valid, doesn't it commit to a single possible blob (let's say that of length = blob_length)? Or are you saying the payload is badly encoded inside the blob?
2. wdym by "proxy return such invalid encoding cert"? Is it the cert or blob that is invalid

Is the case you're saying just "cert is valid, but proxy is unable to download/fetch (from relay and validators) a blob that matches the commitment)? I think that should just return a 404 error no?